### PR TITLE
Update show hide

### PR DESCRIPTION
**Pull request by msmanoj**
## Description
- Updated the ShowHide component to accept a single `className` prop

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1664

## Deployment URL(s)
http://update-show-hide.review.ensembl.org

## Views affected

- External references
- GB drawer sequence download
- EV transcripts download and more information
- Help page `Show me what to do`

### CODE REVIEW

File: src/shared/components/show-hide/ShowHide.scss [hunk 1]
```diff
@@ -1,15 +1,15 @@
 @import 'src/styles/common';
 
 .showHide {
-  // increase specificity to be sure to override chevron's own styles 
+  // increase specificity to be sure to override chevron's own styles
   .chevron {
     margin-left: 10px;
-    height: 6px;
+    --chevron-height: 6px;
```
Comment (azangru) [inline src/shared/components/show-hide/ShowHide.scss:7]: We want this property to be overridable from the parent as well:

```
--chevron-height: var(--show-hide-chevron-height, 6px);
```

File: src/shared/components/show-hide/ShowHide.scss [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   }
-}
 
-.label {
-  color: $blue;
-  cursor: pointer;
-  white-space: nowrap;
+  .label {
+    color: var(--show-hide-label-color, $blue);
+    cursor: var(--show-hide-label-cursor, pointer);
```
Comment (azangru) [inline src/shared/components/show-hide/ShowHide.scss:12]: Is `cursor` something that needs configuring? In `ShowHide.tsx`, doesn't a click on the label (if the label exist) always toggle the show/hide?

File: src/shared/components/show-hide/ShowHide.tsx [hunk 3]
```diff
@@ -30,33 +30,23 @@ import styles from './ShowHide.scss';
...
-    svg: classNames(styles.chevron, props.classNames?.chevron)
-  };
+  const wrapperClasses = classNames(styles.showHide, props.className);
 
   return (
     <div className={wrapperClasses}>
-      <span onClick={props.onClick} className={labelClasses}>
-        {props.label}
+      <span onClick={props.onClick} className={styles.label}>
+        {props.label && props.label}
```
Comment (azangru) [inline src/shared/components/show-hide/ShowHide.tsx:45]: This looks odd. Is the `&& props.label` even required?

Comment (azangru) [reply src/shared/components/show-hide/ShowHide.tsx:45]: BTW, don't we want to change the structure of the ShowHide component to something like this:


```ts
<div onClick={props.onClick} className={wrapperClasses}>
  {props.label && (
    <span className={styles.label}>
      {props.label}
    </span>  
  )}
  <Chevron
    direction={props.isExpanded ? 'up' : 'down'}
    animate={true}
    classNames={{ wrapper: styles.chevron }}
  />
</div>
```

i.e. move the onClick onto the outer element, and wrap the label in its own span with its own class?

File: src/shared/components/show-hide/ShowHide.tsx [hunk 4]
```diff
@@ -30,33 +30,23 @@ import styles from './ShowHide.scss';
...
         <Chevron
           direction={props.isExpanded ? 'up' : 'down'}
           animate={true}
-          classNames={chevronClasses}
+          classNames={{ wrapper: styles.chevron }}
```
Comment (azangru) [inline src/shared/components/show-hide/ShowHide.tsx:49]: OMG! The chevron also has a `classNames` property? Can we transform it into a single `className` as well?

File: src/shared/components/show-hide/ShowHide.scss [hunk 5]
```diff
@@ -1,15 +1,15 @@
 @import 'src/styles/common';
 
 .showHide {
-  // increase specificity to be sure to override chevron's own styles 
+  // increase specificity to be sure to override chevron's own styles
   .chevron {
     margin-left: 10px;
-    height: 6px;
```
Comment (ens-ma7) [inline src/shared/components/show-hide/ShowHide.scss:7]: `cursor: pointer` is missing, if you hoover over chevron for showHide (Entityviewer Filter or Entity Viewer more information), the mouse isn't a pointer.

Comment (azangru) [reply src/shared/components/show-hide/ShowHide.scss:7]: Well spotted 👍 I've pushed an update.

Comment (ens-ma7) [reply src/shared/components/show-hide/ShowHide.scss:7]: Its not properly fixed, now the whole area (even the empty space after the chevron) is clickable (mouse pointer)

Comment (azangru) [reply src/shared/components/show-hide/ShowHide.scss:7]: > Its not properly fixed, now the whole area (even the empty space after the chevron) is clickable (mouse pointer)

It's due to the context where the show/hide is in. If you check out the show/hide itself, it behaves correctly as an inline-block element:

https://user-images.githubusercontent.com/6834224/182147655-aa235ec0-4a9b-4d5d-8e0f-0e0d5aac2f2b.mov

The problem that you are seeing is probably on the entity viewer, where the show/hide expands to fill the css grid column. I'll update the styles there.

Comment (azangru) [reply src/shared/components/show-hide/ShowHide.scss:7]: I pushed an update that should prevent the growing of the show/hide container inside the EntityViewer transcript info panels.

**Review by jyothishnt — APPROVED** [review]
Looks good

File: src/content/app/entity-viewer/gene-view/GeneView.tsx [hunk 6]
```diff
@@ -186,9 +184,7 @@ const GeneViewWithData = (props: GeneViewWithDataProps) => {
           {props.gene.transcripts.length > 5 && (
             <div className={styles.filterLabelWrapper}>
               <ShowHide
-                classNames={{
-                  label: filterLabelClass
-                }}
+                className={isFilterPanelOpen ? styles.showHide : undefined}
```
Comment (jyothishnt) [inline src/content/app/entity-viewer/gene-view/GeneView.tsx:187]: Is it better to write undefined or ''? In my PR I used the latter

Comment (azangru) [reply src/content/app/entity-viewer/gene-view/GeneView.tsx:187]: `className={undefined}` will result in no `class` attribute on the DOM element:

```
<span>Hello</span>
```

whereas `className=""` will most likely result in an empty `class` attribute on the DOM element (unless React has changed its behaviour):

```
<span class>Hello</span>
```

I think the first result is cleaner.

**Review by ens-ma7 — APPROVED** [review]

**Pull request merged**